support-frontend/assets/components/subscriptionCheckouts/address/addressFieldsStore.jsx

```diff
@@ -81,6 +81,9 @@ const checkpostCodeLength = (input: string | null): boolean => ((input == null)
 const isStateNullable = (country: Option<IsoCountry>): boolean =>
   country !== 'AU' && country !== 'US' && country !== 'CA';
 
+const checkStateFieldLength = (state: string | null): boolean =>
+  ((state == null) || (state.length <= 40));
```

Comment:
Is it worth making this more reusable, rather than hard-coding the number 40 (and in the error message too)? We might want to validate for different lengths on other inputs, and it would be helpful to pull out the number into a variable with a descriptive name in any case.